ycmd/tests/subcommands_test.py

```diff
@@ -1282,3 +1282,61 @@ def DefinedSubcommands_WorksWhenNoExplicitCompleterTargetSpecified_test():
          'GoTo' ],
        app.post_json( '/defined_subcommands', subcommands_data ).json )
 
+
+@with_setup( Setup )
+def RunCompleterCommand_GetType_TypescriptCompleter_test():
+  app = TestApp( handlers.app )
+
+  filepath = PathToTestFile( 'test.ts' )
+  contents = open( filepath ).read()
+
+  event_data = BuildRequest( filepath = filepath,
+                             filetype = 'typescript',
+                             contents = contents,
+                             event_name = 'BufferVisit' )
+
+  app.post_json( '/event_notification', event_data )
+
+  gettype_data = BuildRequest( completer_target = 'filetype_default',
+                               command_arguments = ['GetType'],
+                               line_num = 11,
+                               column_num = 1,
+                               contents = contents,
+                               filetype = 'typescript',
+                               filepath = filepath )
+
+  eq_( {
+         'message': 'var foo: Foo'
+       },
+       app.post_json( '/run_completer_command', gettype_data ).json )
+
+@with_setup( Setup )
+def RunCompleterCommand_GetType_HasNoType_TypescriptCompleter_test():
+  app = TestApp( handlers.app )
+
+  filepath = PathToTestFile( 'test.ts' )
+  contents = open( filepath ).read()
+
+  event_data = BuildRequest( filepath = filepath,
+                             filetype = 'typescript',
+                             contents = contents,
+                             event_name = 'BufferVisit' )
+
+  app.post_json( '/event_notification', event_data )
+
+  gettype_data = BuildRequest( completer_target = 'filetype_default',
+                               command_arguments = ['GetType'],
+                               line_num = 2,
+                               column_num = 1,
+                               contents = contents,
+                               filetype = 'typescript',
+                               filepath = filepath )
+
+  try:
+    app.post_json( '/run_completer_command', gettype_data ).json
```

Comment:
I think this would be better if it used the `raises` matcher from PyHamcrest: https://github.com/hamcrest/PyHamcrest#predefined-matchers